Mathlib/Topology/Basic.lean

```diff
@@ -940,31 +945,57 @@ theorem clusterPt_iff_ultrafilter {f : Filter X} : ClusterPt x f ↔
     ∃ u : Ultrafilter X, u ≤ f ∧ u ≤ 𝓝 x := by
   simp_rw [ClusterPt, ← le_inf_iff, exists_ultrafilter_iff, inf_comm]
 
-theorem mapClusterPt_def {ι : Type*} (x : X) (F : Filter ι) (u : ι → X) :
-    MapClusterPt x F u ↔ ClusterPt x (map u F) := Iff.rfl
+section MapClusterPt
 
-theorem mapClusterPt_iff {ι : Type*} (x : X) (F : Filter ι) (u : ι → X) :
-    MapClusterPt x F u ↔ ∀ s ∈ 𝓝 x, ∃ᶠ a in F, u a ∈ s := by
-  simp_rw [MapClusterPt, ClusterPt, inf_neBot_iff_frequently_left, frequently_map]
-  rfl
+variable {F : Filter α} {u : α → X} {x : X}
+
+theorem mapClusterPt_def : MapClusterPt x F u ↔ ClusterPt x (map u F) := Iff.rfl
+alias ⟨MapClusterPt.clusterPt, _⟩ := mapClusterPt_def
+
+theorem MapClusterPt.mono {G : Filter α} (h : MapClusterPt x F u) (hle : F ≤ G) :
+    MapClusterPt x G u :=
+  h.clusterPt.mono (map_mono hle)
+
+theorem Filter.Tendsto.comp_mapClusterPt' [TopologicalSpace Y] {f : X → Y} {y : Y}
+    (hf : Tendsto f (𝓝 x ⊓ map u F) (𝓝 y)) (hu : MapClusterPt x F u) : MapClusterPt y F (f ∘ u) :=
+  (tendsto_inf.2 ⟨hf, tendsto_map.mono_left inf_le_right⟩).neBot (hx := hu)
+
+theorem Filter.Tendsto.comp_mapClusterPt [TopologicalSpace Y] {f : X → Y} {y : Y}
+    (hf : Tendsto f (𝓝 x) (𝓝 y)) (hu : MapClusterPt x F u) : MapClusterPt y F (f ∘ u) :=
+  (hf.mono_left inf_le_left).comp_mapClusterPt' hu
+
+theorem ContinuousAt.comp_mapClusterPt [TopologicalSpace Y] {f : X → Y} (hf : ContinuousAt f x)
+    (hu : MapClusterPt x F u) : MapClusterPt (f x) F (f ∘ u) :=
+  Tendsto.comp_mapClusterPt hf hu
```

Comment:
I would expect `ContinuousAt.comp_mapClusterPt` to prove `ContinuousAt _ _`, not `MapClusterPt _ _ _`. What about
```suggestion
theorem MapClusterPt.tendsto_comp' [TopologicalSpace Y] {f : X → Y} {y : Y}
    (hf : Tendsto f (𝓝 x ⊓ map u F) (𝓝 y)) (hu : MapClusterPt x F u) : MapClusterPt y F (f ∘ u) :=
  (tendsto_inf.2 ⟨hf, tendsto_map.mono_left inf_le_right⟩).neBot (hx := hu)

theorem MapClusterPt.tendsto_comp [TopologicalSpace Y] {f : X → Y} {y : Y}
    (hf : Tendsto f (𝓝 x) (𝓝 y)) (hu : MapClusterPt x F u) : MapClusterPt y F (f ∘ u) :=
  (hf.mono_left inf_le_left).comp_mapClusterPt' hu

theorem MapClusterPt.continuousAt_comp [TopologicalSpace Y] {f : X → Y} (hf : ContinuousAt f x)
    (hu : MapClusterPt x F u) : (f x) F (f ∘ u) :=
  Tendsto.comp_mapClusterPt hf hu
```